src/spikeinterface/widgets/unit_spatial.py

```diff
@@ -0,0 +1,152 @@
+from __future__ import annotations
+
+import numpy as np
+from probeinterface import Probe
+from probeinterface.plotting import get_auto_lims
+from seaborn import color_palette
```

Comment:
you should move these to the `plot_matplotlib` function. Also `seaborn` is not installed by default. Could you use a matplotlib palette?

Reply:
I can definitely use a `matplotlib` palette, but I was thinking to add an optional `seaborn` import if available to enable also `seaborn` palettes. I used `seaborn` also in other parts of the function, I will try to fix it, though I am not sure I will be able to easily plot kernel desnity estimates with `matplotlib` only.

Is there a specific reason to move probeinterface imports to the `plot_matplotlib` function? I use its `get_auto_lims` function to compute `xrange` and `yrange`, that are independent on the visualization backend.

Reply:
yes that souns good! The reason are the failing tests ;)

Core tests only install minimal dependencies. Upon collecting tests across modules, if something is not installed it'll throw an error.

Reply:
I checked the core tests. It seems that `probeinterface` is installed, but `matplotlib` is not, thus `probeinterface.plotting` cannot be imported. Is that right?